extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -2339,8 +2339,11 @@ public static function onArticleCommentGetSquidURLs( $title, &$urls ) {
 	 */
 	private static function shouldBoldenFollowedLink( User $user, Title $title, WallMessage $wm ) {
 		$isWatchlist = $title->isSpecial( 'Watchlist' );
-		return ( F::app()->wg->ShowUpdatedMarker && !$user->isAnon()
+		return ( !$user->isAnon()
+			// Is this user watching this comment, or the thread, or the wall/board?
 			&& ( $wm->isWatched( $user ) || $wm->isWallWatched( $user ) || $wm->isWallOwner( $user ) )
+			// If yes, are we on RecentChanges, or was the page updated?
+			&& ( !$isWatchlist || $wm->getTitle()->getNotificationTimestamp( $user ) || $wm->getArticleTitle()->getNotificationTimestamp( $user ) )
```

Comment:
I think this might be easier to follow if each `&&` set were broken out into separate conditionals, e.g. from:
```php
return ( A && B && C );
```
to 
```php
if ( !A ) {
  return false;
}
if ( !B ) {
  return false;
}
if ( !C ) {
  return false;
}
return true;
```
Additionally, anything loaded from `$wm`, e.g. the `$wm->getTitle()` from `$wm->getTitle()->getNotificationTimestamp( $user )` should be tested first.  Wall code is notorious for having stale IDs, so its very likely that `$wm->getTitle()` could return `null` instead of a title and cause a runtime error.  If you follow my first suggestion you could do this easily by noting that in the last `C` conditional can be written as:
```php
if ( !( C1 || C2 || C3) ) {
  return false;
}
return true;
```
and is the same as;
```php
if ( C1 ) {
  return true;
}
if ( C2 ) {
  return true;
}
if ( C3 ) {
  return true;
}
return false
```
Expanding it this way would let you add the test for null objects easily and clearly:
```php
if ( $wm->getTitle() && $wm->getTitle()->getNotificationTimestamp( $user ) ) {
  return true;
}
```
This makes the method longer, but I think ultimately more readable.